lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -74,10 +51,27 @@ def exists?
 
     private
 
-    def pod_probably_new?(pod_json)
-      return false unless @deploy_started
-      # Shitty, brittle workaround to identify a pod from the new ReplicaSet before implementing ReplicaSet awareness
-      Time.parse(pod_json["metadata"]["creationTimestamp"]) >= (@deploy_started - 30.seconds)
+    def find_latest_rs(deployment_data)
+      label_string = deployment_data["spec"]["selector"]["matchLabels"].map { |k, v| "#{k}=#{v}" }.join(",")
+      raw_json, _err, st = kubectl.run("get", "replicasets", "--output=json", "--selector=#{label_string}")
+      return unless st.success?
+
+      all_rs_data = JSON.parse(raw_json)["items"]
+      current_revision = deployment_data["metadata"]["annotations"]["deployment.kubernetes.io/revision"]
+
+      latest_rs_data = all_rs_data.find do |rs|
+        rs["metadata"]["ownerReferences"].any? { |ref| ref["uid"] == deployment_data["metadata"]["uid"] } &&
+        rs["metadata"]["annotations"]["deployment.kubernetes.io/revision"] == current_revision
+      end
+
+      ReplicaSet.new(
+        name: latest_rs_data["metadata"]["name"],
+        namespace: namespace,
+        context: context,
+        parent: "#{@name.capitalize} deployment",
+        logger: @logger,
+        deploy_started: @deploy_started
+      ).tap { |rs| rs.sync(latest_rs_data) }
```

Comment:
I can also make a method like this if you still think it is better:
```ruby
    def self.from_json(**args)
      json_data = args.delete(:json_data)
      inst = new(**args)
      inst.sync(json_data)
      inst
    end
```
I'm not sure it adds much value, since most of the initializer params aren't from the json, but I'm not against it.